runhouse/resources/hardware/utils.py

```diff
@@ -410,15 +407,28 @@ def parse_filters(since: str, cluster_status: str):
 def get_clusters_from_den(cluster_filters: dict):
     get_clusters_params = {"resource_type": "cluster", "folder": rns_client.username}
 
+    if "cluster_status" in cluster_filters:
+        # Include the relevant daemon status for the filter based on the cluster status provided
+        if cluster_filters["cluster_status"] == ClusterStatus.RUNNING:
+            cluster_filters["status"] = RunhouseDaemonStatus.RUNNING
+
+        elif cluster_filters["cluster_status"] == ClusterStatus.TERMINATED:
+            cluster_filters["status"] = RunhouseDaemonStatus.TERMINATED
+
     # send den request with filters if the user specifies filters.
     # If "all" filter is specified - get all clusters (no filters are added to get_clusters_params)
     if cluster_filters and "all" not in cluster_filters.keys():
         get_clusters_params.update(cluster_filters)
 
     # If not filters are specified, get only running clusters.
     elif not cluster_filters:
+        # For ondemand clusters, use cluster status (via Sky). for other clusters, use the status of the daemon
         get_clusters_params.update(
-            {"cluster_status": "running", "since": LAST_ACTIVE_AT_TIMEFRAME}
+            {
+                "cluster_status": ClusterStatus.RUNNING,
+                "status": RunhouseDaemonStatus.RUNNING,
```

Comment:
should we call it something like "daemon_status" and be more explicit than "status" here since there's multiple statuses? to avoid potential confusion

Reply:
ya we've been thinking about this, just has a bit of BC implications. @mkandler you think we should make this change now also?